sql: block hash sharded indexes for REGIONAL BY ROW

otan opened the pull request:
This is not supported, so block entry points where a REGIONAL BY ROW
table becomes hash sharded.

Release justification: low risk change to new functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61282)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61282)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/create_table.go, line 1724 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/61282#-MUn8IFkALTCDh-6L-Ba:-MUn8IFkALTCDh-6L-Bb:b-g9n0p4) ([raw file](https://github.com/cockroachdb/cockroach/blob/da1b08be3e346b974eae27ec4902ff470aa9f138/pkg/sql/create_table.go#L1724)):*
> ```Go
> 					return nil, hashShardedIndexesDisabledError
> 				}
> 				if isRegionalByRow {
> ```

Nit: Since we're printing this exact same message out in several different places (and in several files), it might make sense to encapsulate this check and error message into a function.


<!-- Sent from Reviewable.io -->

otan committed:
sql: block hash sharded indexes for REGIONAL BY ROW

This is not supported, so block entry points where a REGIONAL BY ROW
table becomes hash sharded.

Release justification: low risk change to new functionality

Release note: None

otan commented:
bors r=ajstorm

otan commented:
`TestInvalidOwnedDescriptorsAreDroppable` 

not moi!

bors r=ajstorm

otan commented:
bors r=ajstorm

otan commented:
bors r=ajstorm
